elide-standalone/src/main/java/com/yahoo/elide/standalone/config/ElideStandaloneSettings.java

```diff
@@ -74,6 +80,25 @@
         return Collections.emptyMap();
     }
 
+    /**
+     * A Set containing Types to be excluded from EntityDictionary's EntityBinding.
+     * @return Set of Types.
+     */
+    default Set<Type> getEntitiesToExclude() {
+        Set<Type> entitiesToExclude = new HashSet();
+        ElideStandaloneAsyncSettings asyncProperties = getAsyncProperties();
+
+        if (asyncProperties == null || !asyncProperties.enabled()) {
+            entitiesToExclude.add(new ClassType(AsyncQuery.class));
+        }
+
+        if (asyncProperties == null || !asyncProperties.enableExport()) {
```

Comment:
checking `asyncProperties == null` twice but it keeps the logic easy to understand instead of using a nested if else block.